File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] n, int target) {
        int s=0,e=n.length-1;
        while(s<=e){
            int mid=(s+e)/2;
            if (target==n[mid]){
                return mid;

            }
            else if (target>n[mid]) {
                s=mid+1;
            }
            else{
                e=mid-1;
            }
        }
        return s;
        
    }
}

File: 35-search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, n: List[int], t: int) -> int:
        s,e=0,len(n)-1
        while(s<=e):
            mid = int((s+e)/2)
            if(t == n[mid]):
                return mid
            elif t> n[mid]:
                s=mid+1
            else:
                e=mid-1
        return s
        